src/tmt_web/api.py

```diff
@@ -0,0 +1,163 @@
+import os
+from typing import Annotated, Any, Literal
+
+from celery.result import AsyncResult
+from fastapi import FastAPI
+from fastapi.params import Query
+from pydantic import BaseModel
+from starlette.responses import HTMLResponse
+
+from tmt_web import service
+from tmt_web.generators import html_generator
+
+app = FastAPI()
+
+
+class TaskOut(BaseModel):
+    id: str
+    status: str
+    result: str | None = None
+    status_callback_url: str | None = None
+
+
+# Sample url: https://tmt.org/?test-url=https://github.com/teemtee/tmt&test-name=/tests/core/smoke
+# or for plans: https://tmt.org/?plan-url=https://github.com/teemtee/tmt&plan-name=/plans/features/basic
+@app.get("/")
+def find_test(
+        test_url: Annotated[
```

Comment:
I would personally reduce the amount of options.

From:
* `test-url`
* `test-name`
* `test-ref`
* `test-path`
* `plan-url`
* `plan-name`
* `plan-ref`
* `plan-path`

To:
* `git-url`
* `git-ref`
* `metadata-name`
* `metadata-path`
* `metadata-type` (enum allowing `test` and `plan` values)

This would IMO simplify the usage as it would get rid of mutually exclusive options.